src/common/vector_operations/vector_cast.cpp

```diff
@@ -614,7 +614,7 @@ void FillEnum(Vector &source, Vector &result, idx_t count) {
 
 	result.SetVectorType(VectorType::FLAT_VECTOR);
 
-	auto str_vec = EnumType::GetValuesInsertOrder(source.GetType());
+	Vector str_vec(EnumType::GetValuesInsertOrder(source.GetType()));
```

Comment:
No reason to create a new vector here, no? Can we just use a reference to the existing one?

```cpp
auto &str_vec = EnumType::GetValuesInsertOrder(source.GetType());
```